src/util/std_types.cpp

```diff
@@ -18,6 +18,17 @@ Author: Daniel Kroening, kroening@kroening.com
 #include "std_expr.h"
 #include "string2int.h"
 
+void array_typet::check(const typet &type, const validation_modet vm)
+{
+  PRECONDITION(type.id() == ID_array);
+  const array_typet &array_type = static_cast<const array_typet &>(type);
+  if(array_type.size().is_nil())
+  {
+    DATA_CHECK(
+      vm, array_type.size() == nil_exprt{}, "array size must be an expression");
+  }
```

Comment:
The message in that check could be clearer.

Reply:
How about the updated message?